modules/RTC/JitsiRemoteTrack.js

```diff
@@ -40,222 +44,236 @@ export default function JitsiRemoteTrack(
+            this.track.addEventListener('unmute', () => this._onTrackUnmute());
+        } else if (this.track.attachEvent) {
+            // FIXME Internet Explorer is not emitting out mute/unmute events.
+            this.track.attachEvent('onmute', () => this._onTrackMute());
+            this.track.attachEvent('onunmute', () => this._onTrackUnmute());
+        }
     }
-};
-
-/**
- * Callback invoked when the track is muted. Emits an event notifying listeners
- * of the mute event.
- *
- * @private
- * @returns {void}
- */
-JitsiRemoteTrack.prototype._onTrackMute = function() {
-    logger.debug(
-        `"onmute" event(${Date.now()}): `,
-        this.getParticipantId(), this.getType(), this.getSSRC());
-
-    this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_MUTE, this);
-};
 
-/**
- * Callback invoked when the track is unmuted. Emits an event notifying
- * listeners of the mute event.
- *
- * @private
- * @returns {void}
- */
-JitsiRemoteTrack.prototype._onTrackUnmute = function() {
-    logger.debug(
-        `"onunmute" event(${Date.now()}): `,
-        this.getParticipantId(), this.getType(), this.getSSRC());
-
-    this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_UNMUTE, this);
-};
-
-/**
- * Sets current muted status and fires an events for the change.
- * @param value the muted status.
- */
-JitsiRemoteTrack.prototype.setMute = function(value) {
-    if (this.muted === value) {
-        return;
+    /**
+     * Callback invoked when the track is muted. Emits an event notifying
+     * listeners of the mute event.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onTrackMute() {
+        logger.debug(
+            `"onmute" event(${Date.now()}): `,
+            this.getParticipantId(), this.getType(), this.getSSRC());
+
+        this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_MUTE, this);
     }
 
-    if (value) {
-        this.hasBeenMuted = true;
+    /**
+     * Callback invoked when the track is unmuted. Emits an event notifying
+     * listeners of the mute event.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onTrackUnmute() {
+        logger.debug(
+            `"onunmute" event(${Date.now()}): `,
+            this.getParticipantId(), this.getType(), this.getSSRC());
+
+        this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_UNMUTE, this);
     }
 
-    // we can have a fake video stream
-    if (this.stream) {
-        this.stream.muted = value;
+    /**
+     * Sets current muted status and fires an events for the change.
+     * @param value the muted status.
+     */
+    setMute(value) {
+        if (this.muted === value) {
+            return;
+        }
+
+        if (value) {
+            this.hasBeenMuted = true;
+        }
+
+        // we can have a fake video stream
+        if (this.stream) {
+            this.stream.muted = value;
+        }
+
+        this.muted = value;
+        this.eventEmitter.emit(JitsiTrackEvents.TRACK_MUTE_CHANGED, this);
```

Comment:
Do these these have a reference to this.eventEmitter? Would these calls be to this.emit?